test/api-test.feature

```diff
@@ -0,0 +1,281 @@
+      | 'https://' + path + '/channel/journal?query=Sensors'               |
+      | 'https://' + path + '/channel/journal/2019/' + existingJournal     |
+      | 'https://' + path + '/channel/publisher?query=Taylor'              |
+      | 'https://' + path + '/channel/publisher/2019/' + existingPublisher |
+
+  Scenario Outline: Slow upstream gives Gateway Timeout problem response
+    * configure headers = { 'Accept': JSON_LD_MEDIA_TYPE, 'Authorization: Basic ' + token }
+    * contentType = responseHeaders['Content-Type'][0]
+    Given url <VALID_URL>
+    And the Channel Register response takes longer than 2 seconds
+    When method get
+    Then status 504
+    And match contentType == PROBLEM_JSON_MEDIA_TYPE
+    And match response.title == 'Gateway Timeout'
+    And match response.status == 504
+    And match response.detail == 'Your request cannot be processed at this time because the upstream server response took too long'
+    And match response.instance == <VALID_URL>
+    And match response.requestId == '#notnull'
+
+    Examples:
+      | VALID_URL                                                          |
+      | 'https://' + path + '/channel/journal?query=Sensors'               |
+      | 'https://' + path + '/channel/journal/2019/' + existingJournal     |
+      | 'https://' + path + '/channel/publisher?query=Taylor'              |
+      | 'https://' + path + '/channel/publisher/2019/' + existingPublisher |
+
+  Scenario Outline: Unexpected error returns Internal Server Error problem response
+    * configure headers = { 'Accept': JSON_LD_MEDIA_TYPE, 'Authorization: Basic ' + token }
+    * contentType = responseHeaders['Content-Type'][0]
+    Given url <VALID_URL>
+    When method get
+    And the channels API application experiences an unexpected error
+    Then status 500
+    And match contentType == PROBLEM_JSON_MEDIA_TYPE
+    And match response.title == 'Internal Server Error'
+    And match response.status == 500
+    And match response.detail == 'Your request cannot be processed at this time because of an internal server error'
+    And match response.instance == <VALID_URL>
+    And match response.requestId == '#notnull'
+
+    Examples:
+      | VALID_URL                                                          |
+      | 'https://' + path + '/channel/journal?query=Sensors'               |
+      | 'https://' + path + '/channel/journal/2019/' + existingJournal     |
+      | 'https://' + path + '/channel/publisher?query=Taylor'              |
+      | 'https://' + path + '/channel/publisher/2019/' + existingPublisher |
+
+  Scenario Outline: Query with unacceptable method returns Not acceptable error
+    * configure headers = { 'Accept': JSON_LD_MEDIA_TYPE, 'Authorization: Basic ' + token }
+    * contentType = responseHeaders['Content-Type'][0]
+    Given url 'https://' + path + '/channel/journal?query=Sensors'
+    When method <METHOD>
+    Then status 406
```

Comment:
From Mozilla's documentation I think this is a 405- Method-not-allowed. (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405)